// @starknet-react/chains/dist/index.js
// src/starknet.ts
var mainnet = {
  id: BigInt("0x534e5f4d41494e"),
  network: "mainnet",
  name: "Starknet",
  nativeCurrency: {
    address: "0x049d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7",
    name: "Ether",
    symbol: "ETH",
    decimals: 18
  },
  rpcUrls: {
    alchemy: {
      http: ["https://starknet-mainnet.g.alchemy.com/v2"]
    },
    infura: {
      http: ["https://starknet-mainnet.infura.io/v3"]
    },
    lava: {
      http: ["https://g.w.lavanet.xyz:443/gateway/strk/rpc-http"]
    },
    default: {
      http: []
    },
    public: {
      http: ["https://rpc.starknet.lava.build"]
    }
  }
};
var goerli = {
  id: BigInt("0x534e5f474f45524c49"),
  network: "goerli",
  name: "Starknet Testnet",
  nativeCurrency: {
    address: "0x049d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7",
    name: "Ether",
    symbol: "ETH",
    decimals: 18
  },
  testnet: true,
  rpcUrls: {
    alchemy: {
      http: ["https://starknet-goerli.g.alchemy.com/v2"]
    },
    infura: {
      http: ["https://starknet-goerli.infura.io/v3"]
    },
    lava: {
      http: ["https://g.w.lavanet.xyz:443/gateway/strkt/rpc-http"]
    },
    default: {
      http: []
    },
    public: {
      http: ["https://rpc.starknet-testnet.lava.build"]
    }
  }
};

// src/devnet.ts
var devnet = {
  id: BigInt("0x534e5f474f45524c49"),
  network: "devnet",
  name: "Starknet Devnet",
  nativeCurrency: {
    address: "0x049d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7",
    name: "Ether",
    symbol: "ETH",
    decimals: 18
  },
  testnet: true,
  rpcUrls: {
    default: {
      http: []
    },
    public: {
      http: ["http://localhost:5050/rpc"]
    }
  }
};
export {
  devnet,
  goerli,
  mainnet
};
